### ISSUE TITLE
[BUG] Running test via VSCode with "Show Browser" using setup dependency shows error running test

### ISSUE THREAD

**Issue by alisterscott** [microsoft/playwright#22051]
### System info
- Playwright Version: 1.32.1
- Operating System: macOS 12.6.3
- Browser: All
- Other info:

### Source code

**Config file**

```js
  projects: [
    { name: 'setup', testMatch: /auth\.setup\.ts/ },
    {
      name: 'main tests',
      use: {
        ...devices['Chromium'],
        storageState: 'playwright/.auth/admin.json',
      },
      testDir: './tests', 
      dependencies: ['setup'],
    }]
```

![image](https://user-images.githubusercontent.com/128826/228414148-d353f53c-5076-422f-a2d9-c912b5529d75.png)

**Steps**
- tick "show browser" in VSCode
- Run the test

**Expected**

Works as per show browser not set

**Actual**

```
    Error: browserType.connect: WebSocket error: connect ECONNREFUSED 127.0.0.1:60054
    =========================== logs ===========================
    <ws connecting> ws://127.0.0.1:60054/e79324c018f1f82bb758d495bb34451a
    <ws error> ws://127.0.0.1:60054/e79324c018f1f82bb758d495bb34451a error connect ECONNREFUSED 127.0.0.1:60054
    <ws connect error> ws://127.0.0.1:60054/e79324c018f1f82bb758d495bb34451a connect ECONNREFUSED 127.0.0.1:60054
    <ws disconnected> ws://127.0.0.1:60054/e79324c018f1f82bb758d495bb34451a code=1006 reason=
    ============================================================
```

**Comment by yury-s** [comment]
@alisterscott can yo share complete project where this happens so that we can reproduce it locally? I tried with a simple setup and couldn't reproduce it.

**Comment by alisterscott** [comment]
> Do you call browserType.connect in your test?

No

This error seems specific to our drupal site as I can't reproduce with generic test case 😕

Adding `await page.waitForTimeout(3000);` to the end of the auth setup code causes this issue to not happen

I'll close this as can't reproduce with generic test case

**Comment by niekbruins** [comment]
> > Do you call browserType.connect in your test?
> 
> No
> 
> This error seems specific to our drupal site as I can't reproduce with generic test case confused
> 
> Adding `await page.waitForTimeout(3000);` to the end of the auth setup code causes this issue to not happen
> 
> I'll close this as can't reproduce with generic test case

I had the same error. 

Instead of `await page.waitForTimeout(3000);` i use `await page.waitForLoadState('networkidle');` 
Which solved the problem in my case.

**Comment by alisterscott** [comment]
@niekbruins thanks for suggestion, I tried that in our app both before and after saving the local storage but still seeing the `Error: browserType.connect: Browser has been closed` error without the sleep.

**Comment by thalfeld** [comment]
Had same issue - downgrading the VSCode extension from 1.0.10 -> 1.0.9 and restarting VSCode (not reloading window) solved it.
`await page.waitForLoadState('networkidle');` also solved it when using 1.0.10.

Note: my specific application is using websockets

**Comment by alisterscott** [comment]
Seems to be fixed in 1.32.3

**Comment by justintoman** [comment]
I'm running into this in `@playwright/test@1.40.1` and VS Code extension `v1.0.20`

```ts
const setup = test.extend<LoginFixture>({
  loginPage: async ({page}, use) => {
    await use({
      login: async user => {
        await page.goto('./');
        await page.getByLabel(/username or email/i).fill(user.username);
        await page.getByLabel(/password/i).fill(user.password);
        await page.getByLabel(/password/i).press('Enter');
        await page.waitForURL('./home');
        await page.context().storageState({path: user.storage});
      }
    });
  }
});
```

That is used to log in with 4 different user roles and save their storage state to the file system. Half of the roles error with:

```
Error: Target page, context or browser has been closed
Call log:
  - navigating to "about:blank", waiting until "load"
```

This works fine in headless mode or with `npx playwright test --ui`. The VS code extension is the only thing that crashes.

I've tried swapping the `page.waitForURL('./home')` with other things such as `page.waitForLoadState('networkidle')` but they all have the same error.
